src/styles/main.scss

```diff
@@ -0,0 +1,51 @@
+$days: 31;
+$month-length: 28, 29, 30, 31;
+$start-day: mon, thu, wed, thu, fri, sat, sun;
+$gap: 1px;
+$cell-size: 100px;
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: 100vh;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+@for $i from 1 through $days {
+  .calendar__day:nth-child(#{$i})::before {
+    content: '#{$i}';
+  }
+}
+
+.calendar {
+  gap: $gap;
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * ($cell-size + $gap) - $gap;
+
+  &__day {
+    margin: 0;
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    &:hover {
+      transition-duration: 0.5s;
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+    }
+  }
+}
+
+.calendar__day:first-child {
```

Comment:
The left margin for `.calendar__day:first-child` is hardcoded with a specific value (`606px`). It would be better to calculate this value based on `$cell-size`, `$gap`, and the actual start day of the month to ensure flexibility and maintainability. Consider creating a loop or a function to set the correct margin based on the start day.